packer/instance-scripts/init-bootnode.sh

```diff
@@ -74,10 +78,18 @@ wait_for_terraform_provisioners
 # Get metadata for this instance
 INDEX=$(cat /opt/quorum/info/index.txt)
 AWS_REGION=$(cat /opt/quorum/info/aws-region.txt)
-PUBLIC_IP=$(wait_for_successful_command 'curl http://169.254.169.254/latest/meta-data/public-ipv4')
-HOSTNAME=$(wait_for_successful_command 'curl http://169.254.169.254/latest/meta-data/public-hostname')
+LB_DNS=$(cat /opt/quorum/info/lb_dns.txt)
 BOOT_PORT=30301
 
+# Fetching HOSTNAME
+# Old HOSTNAME=$(wait_for_successful_command 'curl http://169.254.169.254/latest/meta-data/public-hostname')
+# New: hostname should be contained in the LB_DNS
+
+# Fetching PUBLIC_IP
+# Old PUBLIC_IP=$(wait_for_successful_command 'curl http://169.254.169.254/latest/meta-data/public-ipv4')
+# Naive: PUBLIC_IP=$(dig +short $LB_DNS)
+# Issue: Searching from within same VPC might return private IP instead of public
```

Comment:
You can test this if you want. A private IP will start with 10 if I'm not mistaken.